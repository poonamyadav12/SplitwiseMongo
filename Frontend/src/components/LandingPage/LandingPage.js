import React from "react";
import { Button, Col, Container, Image, Row } from "react-bootstrap";
import { connect } from "react-redux";
import { Redirect } from "react-router";

class LandingPage extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return <>
      {this.props.user && <Redirect to="/home" />}
      <Container>
        <Row>
          <Col sm={5}>
            <h1>Less stress when</h1>
            <h1>sharing expenses</h1>
            <h1 style={{ color: 'magenta', marginBottom: '8rem' }}>with housemates.</h1>
            <h3>Keep track of your shared expenses and balances with housemates, trips, groups, friends, and family.</h3>
            <Button href="/signup" variant="success">
              SIGN UP
          </Button>
          </Col>
          <Col sm={6}>
            <Image src="https://splitwise-images.s3.amazonaws.com/images/1616199002433" />
          </Col>
        </Row>
      </Container>
    </>;
  }
}

function mapState(state) {
  const { user } = state.authentication;
  const { registering } = state.registration;
  return { registering, user };
}

const connectedLandingPage = connect(mapState, {})(LandingPage);
export { connectedLandingPage as LandingPage, LandingPage as LandingPageForTest };
