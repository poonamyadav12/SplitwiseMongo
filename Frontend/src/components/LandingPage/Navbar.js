import React, { Component } from 'react';
import { ListGroup, NavDropdown } from 'react-bootstrap';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { userActions } from '../../_actions';
import { ProfileAvatar } from '../Shared/Shared';
import { GrGroup } from 'react-icons/gr';

//create the Navbar Component
class Navbar extends Component {
  constructor(props) {
    super(props);
    this.handleLogout = this.handleLogout.bind(this);
  }
  //handle logout to destroy the cookie
  handleLogout = () => {
    this.props.logout();
  }

  render() {
    return (
      <>
        <div>
          <nav className='navbar navbar-inverse'>
            <div className='container-fluid'>
              <div className='navbar-header'>
                <Link to="/" className='navbar-brand'>
                  Splitwise App
                    </Link>
              </div>
              <ul className='nav navbar-nav'></ul>
              {!this.props.user && (
                <ul className='nav navbar-nav navbar-right'>
                  <li>
                    <Link to='/signup'>
                      <span className='glyphicon glyphicon-user'></span>Sign
                          me Up
                        </Link>
                  </li>
                </ul>
              )}
              <ul className='nav navbar-nav navbar-right'>
                {this.props.user ? (
                  <>
                    <li>
                      <Link to='/groups'>
                        My groups
                      </Link>
                    </li>
                    <li>
                      <NavDropdown
                        title={
                          <ProfileAvatar
                            user={this.props.user}
                            textColor={'white'}
                          />
                        }
                        id='basic-nav-dropdown'
                      >
                        <ListGroup.Item
                          style={{ backgroundColor: 'lightgray' }}
                        >
                          <Link to='/profile'>Profile</Link>
                        </ListGroup.Item>
                        <ListGroup.Item
                          style={{ backgroundColor: 'lightgray' }}
                          onClick={this.handleLogout}
                        >
                          <Link to='#'>Logout</Link>
                        </ListGroup.Item>
                      </NavDropdown>
                    </li>
                  </>
                ) : (
                  <li>
                    <Link to='/login'>
                      <span className='glyphicon glyphicon-log-in'></span>{' '}
                          Login
                        </Link>
                        )
                  </li>
                )}
              </ul>
            </div>
          </nav>
        </div>
      </>
    );
  }
}

function mapState(state) {
  const { user } = state.authentication;
  return { user };
}

const actionCreators = {
  logout: userActions.logout
};

const connectedNavBar = connect(mapState, actionCreators)(Navbar);
export { connectedNavBar as Navbar };