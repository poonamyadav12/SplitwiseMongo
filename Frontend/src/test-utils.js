// test-utils.js
import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { Provider } from 'react-redux'
// Import your own reducer
import rootReducer from '../src/reducers';
import { createLogger } from 'redux-logger';
import { createStore, applyMiddleware, compose } from 'redux';
import thunkMiddleware from 'redux-thunk';

jest.mock('react-router-dom'); // <- Use this line
jest.mock('react-avatar-editor');
jest.mock('axios');

const loggerMiddleware = createLogger();
const composeEnhancers = (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

function render(
  ui,
  {
    initialState,
    store = createStore(
      rootReducer,
      composeEnhancers(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        ))
    ),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }