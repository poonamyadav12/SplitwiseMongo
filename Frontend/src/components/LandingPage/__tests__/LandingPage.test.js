import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import mockAxios from "axios";
import { setupServer } from 'msw/node';
import renderer from 'react-test-renderer'
import React from 'react';
import { fireEvent, render, screen } from '../../../test-utils';
import { LandingPage, LandingPageForTest } from '../LandingPage';


const server = setupServer()

beforeAll(() => server.listen())
afterEach(() => server.resetHandlers())
afterAll(() => server.close())

test('displays landing page', async () => {
  render(<LandingPage />)

  const email = screen.getByText('SIGN UP');
  fireEvent.click(email);
});

test('renders correctly', () => {
  const tree = renderer
    .create(<LandingPageForTest />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
