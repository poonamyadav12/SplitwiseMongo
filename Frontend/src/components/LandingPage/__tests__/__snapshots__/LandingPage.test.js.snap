// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-sm-5"
    >
      <h1>
        Less stress when
      </h1>
      <h1>
        sharing expenses
      </h1>
      <h1
        style={
          Object {
            "color": "magenta",
            "marginBottom": "8rem",
          }
        }
      >
        with housemates.
      </h1>
      <h3>
        Keep track of your shared expenses and balances with housemates, trips, groups, friends, and family.
      </h3>
      <a
        className="btn btn-success"
        href="/signup"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        SIGN UP
      </a>
    </div>
    <div
      className="col-sm-6"
    >
      <img
        className=""
        src="https://splitwise-images.s3.amazonaws.com/images/1616199002433"
      />
    </div>
  </div>
</div>
`;
